pkg/kubecost/allocationprops.go

```diff
@@ -74,6 +74,7 @@ type AllocationProperties struct {
 	ProviderID     string                `json:"providerID,omitempty"`
 	Labels         AllocationLabels      `json:"allocationLabels,omitempty"`
 	Annotations    AllocationAnnotations `json:"allocationAnnotations,omitempty"`
+	PVBreakdown    map[string]PVUsage    `json:"pvBreakDown,omitempty"`
```

Comment:
Is the idea here that PVCost can go away, even? Is in, could we do something like:
```go
func (a *Allocation) PVCost() float64 {
    cost := 0.0
    for _, pv := range a.Properties.PVBreakdown {
        cost += pv.Cost
    }
    return cost
}
```
Or am I misunderstanding what this does?

Reply:
Also, I would argue this belongs on `Allocation`, not on `AllocationProperties`, if it is storing allocation data as opposed to metadata. How would you feel about that change?

Reply:
In general (performance not withstanding) I do like the idea of breaking down PV data on each Allocation for the sake of detail.

Reply:
Yeah definitely, I did the same for PVByteHours(). I think this is an improvement but there are some api repercussions that I am unsure about.

Reply:
Looks like the refactor did cause some memory issues, cost-analyser pod is now failing to schedule. and it seems to be caused by the pod exceeding memory limits, I am also seeing a lot of failed health checks to prometheus

Reply:
On your cluster or on the scale cluster? That seems pretty strange tbh for such a small cluster. Have you confirmed that these changes are the cause of that? Can you resolve the failing-to-schedule by using a different image?

Reply:
Yes I have an image from Tuesday afternoon before the refactor that schedules and works correctly. I am honestly still not 100% sure what is going on with it.